lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -929,10 +933,24 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
                         skeleton
                     ] `Coin.difference` boringFee
                 , computeSelectionLimit = \_ -> NoLimit
-                , maximumCollateralInputCount =
-                    intCast @Word16 @Int $ view #maximumCollateralInputCount pp
-                , minimumCollateralPercentage =
-                    view #minimumCollateralPercentage pp
+                , maximumCollateralInputCount = unsafeIntCast @Natural @Int $
```

Comment:
I think this is fine for now, but in future it might be possible to avoid this if we adjust the type of `maximumCollateralInputCount` to be a `Natural`. However, that would require further surgery that's outside the scope of this PR.

Reply:
Merging as it's not a regression and we can improve further later.